wordle-web/pages/game.vue

```diff
@@ -1,105 +1,215 @@
+                    v-model="username"
+                    outlined
+                    counter
+                    maxlength="25"
+                  ></v-text-field>
 
+                  <v-divider></v-divider>
+
+                  <v-card-actions>
+                    <v-spacer></v-spacer>
+                    <v-btn
+                      color="primary"
+                      text
+                      @click="
+                        toggleDialog()
+                        nameUpdate()
+                        evaluateName()
+                      "
+                    >
+                      Save
+                    </v-btn>
+                  </v-card-actions>
+                </v-form>
+              </v-card-text>
+            </v-card>
+          </v-dialog>
+        </div>
+      </v-col>
+    </v-row>
+
+    <!-- <v-row justify="center"> Game Time: {{ gameTimeSeconds }} </v-row> -->
+
+    <v-alert
+      v-if="wordleGame.gameOver"
+      prominent
+      max-width="80%"
+      :type="gameResult.type"
+    >
       <v-row justify="center">
-        <game-board :wordleGame="wordleGame" />
-      </v-row>
-      <v-row justify="center">
-        <keyboard :wordleGame="wordleGame" />
+        <v-col>
+          <p>{{ gameResult.text }} <br /></p>
+        </v-col>
       </v-row>
-    </v-container>
+      <v-row>
+        <v-btn class="ml-2" :disabled="wordleGame.hasLost" @click="submitScore">
+          Submit Score
+        </v-btn>
+        <v-btn class="ml-2" @click="resetGame"> Play Again </v-btn></v-row
+      >
+    </v-alert>
+
+    <v-row justify="center"
+      ><v-banner v-model="submittedSuccess" color="success" rounded single-line>
+        Your Score has been submitted!
+      </v-banner></v-row
+    >
+
+    <v-row justify="center">
+      <game-board :wordleGame="wordleGame" />
+    </v-row>
+    <v-row justify="center">
+      <keyboard :wordleGame="wordleGame" />
+    </v-row>
   </v-container>
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Watch } from 'vue-property-decorator'
 import { WordsService } from '~/scripts/wordsService'
 import { GameState, WordleGame } from '~/scripts/wordleGame'
 import KeyBoard from '@/components/keyboard.vue'
 import GameBoard from '@/components/game-board.vue'
 import { Word } from '~/scripts/word'
+import { Stopwatch } from 'ts-stopwatch'
 
 @Component({ components: { KeyBoard, GameBoard } })
 export default class Game extends Vue {
   word: string = WordsService.getRandomWord()
   wordleGame = new WordleGame(this.word)
-
-  isLoaded: boolean = false
+  username: string = this.evaluateName()
+  dialog: boolean = false
+  submittedSuccess: boolean = false
+  gameTimeSeconds: number = 0
+  stopwatch = new Stopwatch()
+  rules = {
+    required: (value: any) => !!value || 'Required.',
+    counter: (value: string | any[]) =>
+      value.length <= 25 || 'Max 25 characters',
+  }
```

Comment:
I like this implementation for limiting the max characters in the name. The display under the input box is also nice.